File: python/src/wis/auth/refresh_token.py
import requests
import json
# #############################################
from ..wis_constants import wis_base_url
from ..wis_exception import WISException
from .token import Token
# #############################################

def refresh_token(token: Token) -> Token:
    """ Refresh Token
        Returns a new token object

        Inputs: 
            token (Token):      ...
        
        Returns:
            token (Token):      ...
    """

    url = wis_base_url + "/token"
    payload = json.dumps({ "refreshToken": token.refresh_token })
    headers = {
        ** token.header,
        'Content-Type': 'application/json'
    }

    response = requests.request("POST", url, headers=headers, data=payload)

    if response.status_code == 200:
        return Token(**response.json())
    else:
        raise WISException.from_response(response)
# #############################################

File: typescript/samples/auth/refreshToken.ts
import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
import { Token, WisError } from "../../types";

export async function refreshToken(
  token: string,
  refreshToken: string
): Promise<Token> {
  const data = JSON.stringify({
    refreshToken: refreshToken,
  });

  const config: AxiosRequestConfig<any> = {
    method: "post",
    url: "https://api.services.wassa.io/token",
    headers: {
      Authorization: "Bearer " + token,
      "Content-Type": "application/json",
    },
    data: data,
  };

  try {
    const response: AxiosResponse<any> = await axios(config);
    return response.data;
  } catch (error: any) {
    throw error as WisError;
  }
}
